heron/spi/src/java/com/twitter/heron/spi/utils/TMasterUtils.java

```diff
@@ -43,20 +44,19 @@ private TMasterUtils() {
    * Communicate with TMaster with command
    *
    * @param command the command requested to TMaster, activate or deactivate.
-   * @return true if the requested command is processed successfully by tmaster
    */
   @VisibleForTesting
-  public static boolean sendToTMaster(String command, String topologyName,
+  public static void sendToTMaster(String command, String topologyName,
                                       SchedulerStateManagerAdaptor stateManager,
-                                      NetworkUtils.TunnelConfig tunnelConfig) {
+                                      NetworkUtils.TunnelConfig tunnelConfig)
+      throws TMasterException {
     // fetch the TMasterLocation for the topology
     LOG.fine("Fetching TMaster location for topology: " + topologyName);
 
     TopologyMaster.TMasterLocation location = stateManager.getTMasterLocation(topologyName);
-    if (location == null) {
-      LOG.severe("Failed to fetch TMaster Location for topology: " + topologyName);
-      return false;
-    }
+    Preconditions.checkNotNull(location, "Failed to fetch TMaster Location for topology: "
```

Comment:
consistency is nice, but does that mean we should put statement `Preconditions.checkNotNull...` inside of a `try...catch` clause. If exception `NullPointerException` gets thrown out, we catch it and throw a `TMasterException` out instead? (and [none of the `checkNotNull` accepts exception in arguments](http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/base/Preconditions.html#checkNotNull(T))]